### PR TITLE
upgrade geth to v1.10.18 to avoid using nightly

### PR DESCRIPTION
**Pull request by lispc**
fixing https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/509

### CODE REVIEW

File: eth-types/src/lib.rs [hunk 1]
```diff
@@ -376,52 +376,29 @@ pub struct GethExecTrace {
...
-        trace[i].memory.0.truncate(mem_sizes[i]);
-    }
-}
-
-impl<'de> Deserialize<'de> for GethExecTrace {
-    fn deserialize<D>(deserializer: D) -> Result<GethExecTrace, D::Error>
-    where
-        D: serde::Deserializer<'de>,
-    {
+impl From<GethExecTraceInternal> for GethExecTrace {
```
Comment (ed255) [inline eth-types/src/lib.rs:379]: Now that we don't need to fix anything in the exec trace, the type `GethExecTraceInternal` is no longer necessary.
We just need to update `GethExecTrace` like this:
```diff
-#[derive(Clone, Debug, Eq, PartialEq)]
+#[derive(Deserialize, Clone, Debug, Eq, PartialEq)]
 pub struct GethExecTrace {
     /// Used gas
     pub gas: Gas,
     /// True when the transaction has failed.
     pub failed: bool,
     /// Return value of execution
+    #[serde(rename = "returnValue")]
     pub return_value: String,
     /// Vector of geth execution steps of the trace.
+    #[serde(rename = "structLogs")]
     pub struct_logs: Vec<GethExecStep>,
 }
```

And remove the type `GethExecTraceInternal`, the traits impl `From<GethExecTraceInternal> for GethExecTrace` and `Deserialize<'de> for GethExecTrace`.
You'll need to update the `deserialize_geth_exec_trace2` test to use `GethExecTrace`

**Review by ed255 — COMMENTED** [review]
Please take a look at my comment about removing code that is no longer necessary.  The rest looks good to me :)

**Review by ed255 — APPROVED** [review]
LGTM!

**Review by DreamWuGit — APPROVED** [review]
LGTM, thanks for the update!

**Comment by DreamWuGit** [comment]
Just be curious about how does geth-utils/go.sum generated ? should be by a tool ?

**Pull request merged**